test-case/test-jack-detection-playback-capture.sh

```
@@ -0,0 +1,164 @@
+    usbrelay_switch "$relay" 0
+
+    for round in $(seq 1 "$round_cnt")
+    do
+        for idx in $(seq 0 $((PIPELINE_COUNT - 1)))
+        do
+            initialize_audio_params "$idx"
+
+            channel=$(func_pipeline_parse_value "$idx" channel)
+            rate=$(func_pipeline_parse_value "$idx" rate)
+            fmt=$(func_pipeline_parse_value "$idx" fmt)
+            dev=$(func_pipeline_parse_value "$idx" dev)
+            snd=$(func_pipeline_parse_value "$idx" snd)
+
+            dlogi "===== Testing: (PCM: $pcm [$dev]<$type>) ====="
+
+            aplay_opts -D"$dev" -r "$rate" -c "$channel" -f "$fmt" -d "$duration" "/dev/zero" -q || {
+                func_lib_lsof_error_dump "$snd"
+                die "aplay on PCM $dev failed."
+            }
+
+            dlogi "Unplug jack audio."
+            usbrelay_switch "$relay" 1
+
+            aplay_opts -D"$dev" -r "$rate" -c "$channel" -f "$fmt" -d "$duration" "/dev/zero" -q || {
+                func_lib_lsof_error_dump "$snd"
+                die "aplay on PCM $dev failed."
+            }
+
+            check_control_switch_state "headset" "off" || {
+                die "unplug headset jack failed."
+            }
+
+            check_control_switch_state "headphone" 'off' || {
+                die "unplug headphone jack failed."
+            }
+
+            dlogi "Plug jack audio."
+            usbrelay_switch "$relay" 0
+
+            aplay_opts -D"$dev" -r "$rate" -c "$channel" -f "$fmt" -d "$duration" "/dev/zero" -q || {
+                func_lib_lsof_error_dump "$snd"
+                die "aplay on PCM $dev failed."
+            }
+
+            check_control_switch_state "headset" "on" || {
+                die "Plug headset jack failed."
+            }
+
+            check_control_switch_state "headphone" "on" || {
+                die "Plug headphone jack failed."
+            }
+        done
+    done
+
+    sof-kernel-log-check.sh "$KERNEL_CHECKPOINT"
```

Maybe check after each iteration to "fail fast"?